File: src/main/java/com/transform/transform/PersonTransformer.java
package com.transform.transform;

import com.transform.input.InputPerson;
import com.transform.output.OutputPerson;

public class PersonTransformer extends AbstractTransformer {

    @Override
    public void transform(InputPerson inputPerson, OutputPerson outputPerson) {
        outputPerson.setName(inputPerson.getLastname());

        super.transform(inputPerson, outputPerson);
    }

}

File: src/main/java/com/transform/transform/PersonTransformer.kt
package com.transform.transform

import com.transform.input.InputPerson
import com.transform.output.OutputPerson

class PersonTransformer : AbstractTransformer() {

    override fun transform(inputPerson: InputPerson, outputPerson: OutputPerson) {
        outputPerson.name = inputPerson.lastName

        next?.transform(inputPerson, outputPerson)
    }

}
